eng/scripts/Save-Package-Namespaces-Property.ps1

```diff
@@ -15,26 +15,59 @@ PackageInfo subdirectory. The whl files are going to be in the subdirectory whic
 is the same as the artifact's name but artifact name in the file's name will have
  underscores instead of dashes.
 
-.PARAMETER ArtifactsList
-The list of artifacts to gather namespaces for, this is only done for libraries that are
-producing docs.
--ArtifactsList ('${{ convertToJson(parameters.Artifacts) }}' | ConvertFrom-Json)
+.PARAMETER RepoRoot
+The root of the directory. This will be used in combination with the paths of of
+the properties within each PackageInfo file to determine the full path to other relevant
+metadata...like ci.yml for instance.
 #>
 [CmdletBinding()]
 Param (
     [Parameter(Mandatory = $True)]
     [string] $ArtifactStagingDirectory,
-    [Parameter(Mandatory=$true)]
-    [array] $ArtifactsList
+    [Parameter(Mandatory = $True)]
+    [string] $RepoRoot
 )
-
-$ArtifactsList = $ArtifactsList | Where-Object -Not "skipPublishDocMs"
+.  (Join-Path $PSScriptRoot ".." common scripts Helpers PSModule-Helpers.ps1)
+Install-ModuleIfNotInstalled "powershell-yaml" "0.4.1" | Import-Module
 
 . (Join-Path $PSScriptRoot ".." common scripts common.ps1)
 
-if (-not $ArtifactsList) {
-    Write-Host "ArtifactsList is empty, nothing to process. This can happen if skipPublishDocMs is set to true for all libraries being built."
-    exit 0
+function ShouldPublish ($ServiceDirectory, $PackageName) {
+    $ciYmlPath = Join-Path $ServiceDirectory "ci.yml"
+
+    Write-Host $PackageName
+
+    if (Test-Path $ciYmlPath)
+    {
+        Write-Host $ciYmlPath
+        $ciYml = ConvertFrom-Yaml (Get-Content $ciYmlPath -Raw)
+
+        if ($ciYml.extends -and $ciYml.extends.parameters -and $ciYml.extends.parameters.Artifacts) {
+            $packagesBuildingDocs = $ciYml.extends.parameters.Artifacts `
+                | Where-Object {
+                    if ($_["skipPublishDocMs"]) {
```

Comment:
follow-up: I think this logic currently works but we probably can simplify it, so it is easier to understand.

Reply:
I think this can be `Where-Object { -not ($_["skipPublishDocMs"] -eq $true) }`

Reply:
or `Where-Object -Property 'skipPublishDocMs' -ne -Value $true`

Reply:
> Where-Object { -not ($_["skipPublishDocMs"] -eq $true) }

Will work. Latter will not because property isn't always present.

Reply:
Where-Object properly compares `undefined` to `-Value`:
```powershell
@(@{x=1},@{x=2;y=$true}) | Where-Object -Property 'y' -ne -Value $true

Name                           Value
----                           -----
x                              1
```